services/utilServices/SitesService.js

```diff
@@ -0,0 +1,133 @@
+const axios = require("axios")
+const Bluebird = require("bluebird")
+const _ = require("lodash")
+
+// Import error
+const GH_MAX_REPO_COUNT = 100
+const ISOMERPAGES_REPO_PAGE_COUNT = process.env.ISOMERPAGES_REPO_PAGE_COUNT || 3
+const ISOMER_GITHUB_ORG_NAME = process.env.GITHUB_ORG_NAME
+const ISOMER_ADMIN_REPOS = [
+  "isomercms-backend",
+  "isomercms-frontend",
+  "isomer-redirection",
+  "isomerpages-template",
+  "isomer-conversion-scripts",
+  "isomer-wysiwyg",
+  "isomer-slackbot",
+  "isomer-tooling",
+  "generate-site",
+  "travisci-scripts",
+  "recommender-train",
+  "editor",
+  "ci-test",
+  "infra",
+  "markdown-helper",
+]
+
+class SitesService {
+  constructor({ gitHubService, configYmlService }) {
+    this.githubService = gitHubService
+    this.configYmlService = configYmlService
+  }
+
+  // timeDiff tells us when a repo was last updated in terms of days (for e.g. 2 days ago,
+  // today)
+  timeDiff(lastUpdated) {
+    const gapInUpdate = new Date().getTime() - new Date(lastUpdated).getTime()
+    const numDaysAgo = Math.floor(gapInUpdate / (1000 * 60 * 60 * 24))
+    // return a message for number of days ago repo was last updated
+    switch (numDaysAgo) {
+      case 0:
+        return "Updated today"
+      case 1:
+        return "Updated 1 day ago"
+      default:
+        return `Updated ${numDaysAgo} days ago`
+    }
+  }
+
+  async getSites({ accessToken }) {
+    const endpoint = `https://api.github.com/orgs/${ISOMER_GITHUB_ORG_NAME}/repos`
+
+    // Simultaneously retrieve all isomerpages repos
+    const paramsArr = []
+    for (let i = 0; i < ISOMERPAGES_REPO_PAGE_COUNT; i += 1) {
+      paramsArr.push({
+        per_page: GH_MAX_REPO_COUNT,
+        sort: "full_name",
+        page: i + 1,
+      })
+    }
+
+    const sites = await Bluebird.map(paramsArr, async (params) => {
+      const resp = await axios.get(endpoint, {
+        params,
+        headers: {
+          Authorization: `token ${accessToken}`,
+          "Content-Type": "application/json",
+        },
```

Comment:
i think i've seen this repeated a ton and it merits its own helper method. might be out of scope (and not that important yet), so maybe we could file an issue?

Reply:
This is used only in `SitesService`, `AuthService` and `GithubService` actually, since this is only relevant when querying Github - I'm wondering if the solution is to create a more general misc service to query endpoints which don't start with `https://api.github.com/repos/`? It felt out of place for these 2 services to have axios calls when the only other place which uses axios is `GithubService`

Reply:
actually you're right - maybe we should just defer the methods that require headers to GHA and make our main services agnostic of the data provider (github) wdyt?

Reply:
Yup i think that makes sense - i'll file an issue to tackle this in a future PR